doc/source/whatsnew/v0.21.1.txt

```diff
@@ -64,6 +64,7 @@ Bug Fixes
 - Bug in ``pd.concat`` when empty and non-empty DataFrames or Series are concatenated (:issue:`18178` :issue:`18187`)
 - Bug in :class:`IntervalIndex` constructor when a list of intervals is passed with non-default ``closed`` (:issue:`18334`)
 - Bug in :meth:`IntervalIndex.copy` when copying and ``IntervalIndex`` with non-default ``closed`` (:issue:`18339`)
+- Bug in :meth:`fillna` maximum recursion depth exceeded in cmp (:issue:`18159`).
```

Comment:
"**where** maximum recursion depth **gets** exceeded"

Also, instead of "cmp", replace it with something more reader-friendly (I presume this is comparison?)